zhaquirks/schneiderelectric/thermostat.py

```diff
@@ -572,5 +579,213 @@ class AttributeDefs(CustomCluster.AttributeDefs):
     .replaces(SEMetering, endpoint_id=5)
     .replaces(SECycleTime)
     .replaces(SEHeatingCoolingOutput)
+    .number(
+        cluster_id=SEMetering.cluster_id,
+        endpoint_id=5,
+        attribute_name=SEMetering.AttributeDefs.se_fixed_load_demand.name,
+        translation_key="fixed_load_demand",
+        fallback_name="Fixed Load Demand",
+        device_class=NumberDeviceClass.POWER,
+        unit=UnitOfPower.WATT,
+        min_value=0,
+        max_value=10000,
+        step=1,
+    )
+    .number(
+        cluster_id=SEUserInterface.cluster_id,
+        endpoint_id=1,
+        attribute_name=SEUserInterface.AttributeDefs.se_brightness.name,
+        translation_key="display_brightness",
+        fallback_name="Display Brightness",
+        # unit="%",
```

Comment:
We should have `PERCENTAGE` as a unit for this. It's just a string: `PERCENTAGE = "%"`, so this should already work when uncommented. Can you confirm?

Reply:
Just tested, and if I uncomment this line, I get the following error:

```
    Failed to create platform entity: <class 'zha.application.platforms.number.NumberConfigurationEntity'> [args=('b0:c7:de:ff:fe:be:38:1f-1', [<zha.zigbee.cluster_handlers.hvac.UserInterfaceClusterHandler object at 0x7fe3642a55e0>], <zha.zigbee.endpoint.Endpoint object at 0x7fe3642a4c80>, <CustomDeviceV2 model='EKO07259' manuf='Schneider Electric' nwk=0xDC1A ieee=b0:c7:de:ff:fe:be:38:1f is_initialized=True> - quirk_applied: True - quirk_or_device_class: zigpy.quirks.v2.CustomDeviceV2 - quirk_id: None), kwargs={'entity_metadata': NumberMetadata(entity_platform=<EntityPlatform.NUMBER: 'number'>, entity_type=<EntityType.CONFIG: 'config'>, cluster_id=516, endpoint_id=1, cluster_type=<ClusterType.Server: 0>, initially_disabled=False, attribute_initialized_from_cache=True, translation_key='display_inactive_brightness', fallback_name='Display inactive brightness', attribute_name='se_inactive_brightness', reporting_config=None, min=0, max=100, step=1, unit='%', mode=None, multiplier=None, device_class=None)}]

Traceback (most recent call last):
  File "/nix/store/2dcvclynic8hvxi9lcfrcsy1k1iwbz1p-python3.12-zha-0.0.39/lib/python3.12/site-packages/zha/application/gateway.py", line 335, in create_platform_entities
    platform_entity = platform_entity_class.create_platform_entity(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/2dcvclynic8hvxi9lcfrcsy1k1iwbz1p-python3.12-zha-0.0.39/lib/python3.12/site-packages/zha/application/platforms/number/__init__.py", line 229, in create_platform_entity
    return cls(unique_id, cluster_handlers, endpoint, device, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/2dcvclynic8hvxi9lcfrcsy1k1iwbz1p-python3.12-zha-0.0.39/lib/python3.12/site-packages/zha/application/platforms/number/__init__.py", line 242, in __init__
    super().__init__(unique_id, cluster_handlers, endpoint, device, **kwargs)
  File "/nix/store/2dcvclynic8hvxi9lcfrcsy1k1iwbz1p-python3.12-zha-0.0.39/lib/python3.12/site-packages/zha/application/platforms/__init__.py", line 297, in __init__
    self._init_from_quirks_metadata(entity_metadata)
  File "/nix/store/2dcvclynic8hvxi9lcfrcsy1k1iwbz1p-python3.12-zha-0.0.39/lib/python3.12/site-packages/zha/application/platforms/number/__init__.py", line 269, in _init_from_quirks_metadata
    self._attr_native_unit_of_measurement = validate_unit(
                                            ^^^^^^^^^^^^^^
  File "/nix/store/2dcvclynic8hvxi9lcfrcsy1k1iwbz1p-python3.12-zha-0.0.39/lib/python3.12/site-packages/zha/units.py", line 167, in validate_unit
    return UNITS_OF_MEASURE[type(external_unit).__name__](external_unit.value)
           ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'str'
```

Reply:
I think, as of now,  unit can't be string since then the code resolves into:

`UNITS_OF_MEASURE[type("%").__name__](external_unit.value)` -> `UNITS_OF_MEASURE['str'](external_unit.value)` -> `KeyError: 'str'`

Reply:
For reference, tracking this ZHA related issue in: https://github.com/zigpy/zha/issues/327